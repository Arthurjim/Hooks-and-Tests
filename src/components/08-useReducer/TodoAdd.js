import React from "react";
import useForm from "../../hooks/useForm";

const TodoAdd = ({addTodo}) => {
    const [{ description }, handleChange, reset] = useForm({
        description: "",
    });
    const handleSubmit = (e) => {
        e.preventDefault();
        if (description.trim().length < 1) {
            return;
        }
        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false,
        };
        addTodo(newTodo)
        reset();

    }
    return (
        <>
            <h4>Agregar TODO</h4>
            <hr />
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    className="form-control"
                    name="description"
                    placeholder="Aprenderr ..."
                    autoComplete="off "
                    onChange={handleChange}
                    value={description}
                    id="input"
                />
                <button
                    type="submit"
                    className="btn btn-outline-primary mt-1 col-12"
                >
                    Agregar
                </button>
            </form>
        </>
    );
};

export default TodoAdd;
