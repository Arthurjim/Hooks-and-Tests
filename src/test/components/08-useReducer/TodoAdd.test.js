import { shallow } from "enzyme"
import TodoAdd from "../../../components/08-useReducer/TodoAdd"

describe('Pruebas en <TodoAdd/>', () => {
    const addTodo = jest.fn()
    const wrapper = shallow(<TodoAdd addTodo={addTodo}/>)
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        
    })
    test('No debe de llamar la fn addTodo', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({preventDefault(){}})
        expect(addTodo).toHaveBeenCalledTimes(0)
    });
    test('Debe de llamar la funcion addTodo', () => {
        
        const value = 'Hacer comida';
        wrapper.find('input').simulate('change', { target: { value,name:'description' } });

        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect( addTodo ).toHaveBeenCalled();
        expect( addTodo ).toHaveBeenCalledWith(expect.any(Object)); // { }
        expect( addTodo ).toHaveBeenCalledWith({
            id:expect.any(Number),
            desc:value,
            done:false,
        }); // { }
        expect(wrapper.find('input').prop('value')).toBe('')
    })
    
    
})
