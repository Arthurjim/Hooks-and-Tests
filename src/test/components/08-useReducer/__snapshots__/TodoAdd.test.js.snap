// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd/> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off "
      className="form-control"
      id="input"
      name="description"
      onChange={[Function]}
      placeholder="Aprenderr ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 col-12"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
